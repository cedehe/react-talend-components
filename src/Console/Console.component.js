import React, { PropTypes } from 'react';
import classnames from 'classnames';
import Log from './Log';
import theme from './Console.scss';

/**
 * @param {object} props react props
 * @example
<Console logs={logs}></Console>
 */
function Console(props) {
	const wrapper = classnames(
		'tc-console',
		theme.wrapper
	);
	return (
		<div className={wrapper}>
			{props.logs.map((log, i) => (
				<Log
					key={i}
					level={log.level || 'info'}
					actions={log.actions || []}
					msg={log.msg}
				/>
			))}
		</div>
	);
}

Console.propTypes = {
	logs: PropTypes.arrayOf(Log.propTypes),
};

export default Console;
