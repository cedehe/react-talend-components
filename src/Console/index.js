import Console from './Console.component';

export default Console;
