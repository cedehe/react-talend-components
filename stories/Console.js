import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { Console } from '../src/index';

const logs = [
	{ msg: 'Hello world, how are you ?' },
	{ msg: 'Something goes wrong here', level: 'error' },
	{
		msg: 'Something goes wrong but can be fixed',
		level: 'warning',
		actions: [
			{
				label: 'fixme',
				onClick: action('fixme'),
				icon: 'fa fa-bug',
			},
		],
	},
];

storiesOf('Console', module)
	.add('default', () => (
		<div>
			<h1>Console</h1>
			<Console logs={logs} />
		</div>
	));
