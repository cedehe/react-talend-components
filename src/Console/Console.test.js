import React from 'react';
import renderer from 'react-test-renderer';

import Console from './Console.component';

jest.mock('react-dom');

const logs = [
	{ msg: 'Hello world, how are you ?' },
	{ msg: 'Something goes wrong here', level: 'error' },
	{
		msg: 'Something goes wrong but can be fixed',
		level: 'warning',
		actions: [
			{
				label: 'fixme',
				onClick: jest.fn(),
				icon: 'fa fa-bug',
			},
		],
	},
];

describe('Console', () => {
	it('should render', () => {
		const wrapper = renderer.create(
			<Console logs={logs} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
