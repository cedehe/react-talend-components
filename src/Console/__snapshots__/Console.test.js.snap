exports[`Console should render 1`] = `
<div
  className="tc-console">
  <div
    className="tc-console-log">
    <p
      className="tc-console-log-msg tc-console-log-msg-info">
      Hello world, how are you ?
      <div
        className="tc-actions btn-group" />
    </p>
  </div>
  <div
    className="tc-console-log">
    <p
      className="tc-console-log-msg tc-console-log-msg-error">
      Something goes wrong here
      <div
        className="tc-actions btn-group" />
    </p>
  </div>
  <div
    className="tc-console-log">
    <p
      className="tc-console-log-msg tc-console-log-msg-warning">
      Something goes wrong but can be fixed
      <div
        className="tc-actions btn-group">
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          role={null}
          type="button">
          <i
            className="fa fa-bug" />
          <span>
            fixme
          </span>
        </button>
      </div>
    </p>
  </div>
</div>
`;
